commands/core/config.drush.inc

```diff
@@ -461,9 +557,438 @@ function drush_config_edit($config_name = '') {
   }
 }
 
+function drush_config_merge($alias = '', $config_label = 'staging') {
+  // Allow the user to provide a specific tracking branch to do live work on.
+  $tracking_branch = drush_get_option('tracking', FALSE);
+  $tracking_branch_is_tmp = FALSE;
+  if (!$tracking_branch) {
+    $tracking_branch = 'drush-live-config-temp';
+    $tracking_branch_is_tmp = TRUE;
+  }
+
+  // Figure out what our base commit is going to be for this operation.
+  $merge_info = array(
+    'base' => drush_get_option('base', FALSE),
+    'comment' => drush_get_option('comment', ''),
+    'commit' => !drush_get_option('no-commit', FALSE),
+    'git-transport' => drush_get_option('git', FALSE),
+    'remote' => drush_get_option('remote', 'origin'),
+    'tool' => drush_get_option('tool', ''),
+    'config-label' => $config_label,
+    'live-site' => $alias,
+    'dev-site' => '@self',
+    'live-config' => $tracking_branch,
+    'dev-config' => 'drush-dev-config-temp',
+    'autodelete-live-config' => $tracking_branch_is_tmp,
+    'autodelete-dev-config' => TRUE,
+  );
+  $result = drush_command_invoke_all_ref('drush_config_merge_info_alter', $merge_info);
+  if (array_search(FALSE, $result, TRUE) !== FALSE) {
+    return FALSE;
+  }
```

Comment:
OK we can proceed with this. I just wanted to communicate that a hook can throw an exception instead of returning false and a similar fate will occur.